Python-packages/covidcast-py/covidcast/plotting.py

```diff
@@ -238,20 +240,40 @@ def _join_county_geo_df(data: pd.DataFrame,
     input_cols = list(data.columns)
     # create state FIPS code in copy, otherwise original gets modified
     data = data.assign(state=[i[:2] for i in data[county_col]])
+    if combine_megacounties:
+        merged = _combine_megacounties(data, county_col, geo_info)
+    else:
+        merged = _distribute_megacounties(data, county_col, geo_info, join_type, input_cols)
+    merged[county_col] = merged.GEOID.combine_first(merged[county_col])
+    merged.rename(columns={"STATEFP": "state_fips"}, inplace=True)
+    return gpd.GeoDataFrame(merged[input_cols + ["state_fips", "geometry"]])
+
+
+def _combine_megacounties(data, county_col, geo_info):
+    merged = data.merge(geo_info, how="left", left_on=county_col, right_on="GEOID", sort=True)
+    missing = set(geo_info.GEOID) - set(data.geo_value)
+    for i, row in merged.iterrows():
+        if row.geo_value.endswith("000"):
```

Comment:
It might be worth creating an `is_megacounty` helper that does `endswith("000")`. It's not the worst thing to repeat a few times in the code, but putting it in a function makes the code more readable

Reply:
good call, added.